test/runner.spec.ts

```diff
@@ -192,5 +69,105 @@ describe('runner', () => {
       })
     ).resolves.not.toThrow();
     expect(executedMigrations).toHaveLength(0);
+    expect(executedMigrations).toMatchSnapshot();
+  });
+
+  it('should execute a basic up migration in nested directories', async () => {
+    const executedMigrations: ExecutedMigrations = [];
+
+    const dbClient = mockMigrationClient(() => executedMigrations);
+
+    await expect(
+      runner({
+        dbClient,
+        migrationsTable: 'pgmigrations',
+        // We use cockroach migrations for now, as they are more simple
+        // We either could mock the migration files later or define specific migrations for unit-testing
+        dir: 'test/migrations-subdir',
+        direction: 'up',
+      })
+    ).resolves.not.toThrow();
+    expect(executedMigrations).toHaveLength(4);
+    expect(executedMigrations).toMatchSnapshot();
   });
 });
+
+type ExecutedMigrations = Array<{ id: number; name: string; run_on: Date }>;
+
+function mockMigrationClient(
```

Comment:
**nitpick (blocking):** I have the personal preference of "use after defined"
Could you please move the up in the file?